src/formatDate.js

```diff
@@ -50,7 +50,41 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitDate = date.split(fromFormat[3]);
+  const fromMp = splitDate.reduce(
+    (acc, nextVal, index) => ({ ...acc,
+      [fromFormat[index].charAt(0)]: nextVal }),
+    {}
+  );
+
+  const resultArray = [];
+
+  for (const key of toFormat) {
+    const keyChar = key.charAt(0);
+
+    switch (keyChar) {
+      case 'D':
+      case 'M':
+        resultArray.push(fromMp[keyChar]);
+        break;
+      case 'Y':
+        if (fromMp[keyChar].length === 4 && key.length === 2) {
+          fromMp[keyChar] = fromMp[keyChar].slice(2);
+        }
+
+        if (fromMp[keyChar].length === 2 && key.length === 4) {
+          if (Number(fromMp[keyChar]) >= 30) {
+            fromMp[keyChar] = '19' + fromMp[keyChar];
+          } else {
+            fromMp[keyChar] = '20' + fromMp[keyChar];
+          }
```

Comment:
I think this is a bit easier to read
```suggestion
          const year = Number(fromMp[keyChar]) >= 30 ? "19" : "20";
          fromMp[keyChar] = `${year}${fromMp[keyChar]}`;
```